### ISSUE TITLE
Making the orchestrator work with arbitrary CLIs

### ISSUE THREAD

**Issue by nbren12** [ai2cm/fv3net#174]
After considering how to configure a baseline simulation, I realized a maintainability issue with the current strategy of the orchestrator not working with general command line interfaces.

To submit a prognostic baseline simulation without an ML model, I would like to use the same [script](https://github.com/VulcanClimateModeling/fv3net/blob/master/workflows/prognostic_c48_run/orchestrate_submit_job.py) as for the prognostic runs. Perhaps the simplest way to do this is to make "model_url" an optional argument.

Unfortunately, I don't think the orchestrator "from" capability will work with "extra_args". It would be nice to do something like this:
```
extra_args:
--sklearn-model-url: {from: train_sklearn_model}
```
Of course, I could write another wrapper which calls orchestrate_submit_job.py, but adding wrapper layers causes a significant maintainability problem. Any changes to the prognostic run CLI would require editing all the wrappers calling it. I also think the indirection with e.g. bash wrappers calling python scripts which submit K8s jobs is making the code pretty complicated and hard to follow.

To sum up, I think we should move towards having 1 command line interface per workflow, and the orchestrator should be expanded to work with most command line interfaces. I don't think this should be too hard. For simplicity, I think it is probably still OK to assume that the output is the last positional argument. I also think "extra_args" could be merged with "input" since IIRC the only real difference is the ordering of the generate command line arguments.

cc @frodre @brianhenn
